docs/yaml/functions/_build_target_base.yaml

```diff
@@ -302,3 +302,13 @@ kwargs:
       "proc-macro" is a special rust procedural macro crate.
 
       "proc-macro" is new in 0.62.0.
+
+  rust_dependency_map:
+    type: dict[str]
+    since: 1.2.0
+    description: |
+      On rust targets this provides a map of library names to the crate name
+      with which it would be available inside the rust code.
+
+      This allows renaming similar to the dependency renaming feature of cargo
+      or `extern crate foo as bar` inside rust code.
```

Comment:
It also needs a release notes, add a file in doc/markdown/snippets/

Reply:
Not sure. With dash it can never be used, but also why would you even write it with dash at that place? Should either be converted or be an error. cargo is silently converting it, so for consistency maybe that would be the best choice?

Reply:
Right, I think it's better to convert automatically. Something like `target.rust_dependency_map.get(d.name, d.name).replace('-', '_')`